src/promptflow/promptflow/batch/_batch_engine.py

```diff
@@ -130,11 +131,20 @@ def run(
             # resolve input data from input dirs and apply inputs mapping
             batch_input_processor = BatchInputsProcessor(self._working_dir, self._flow.inputs, max_lines_count)
             batch_inputs = batch_input_processor.process_batch_inputs(input_dirs, inputs_mapping)
-            # run flow in batch mode
+            # resolve output dir
             output_dir = resolve_dir_to_absolute(self._working_dir, output_dir)
+            # run flow in batch mode
             with _change_working_dir(self._working_dir):
-                batch_result = self._exec_batch(batch_inputs, run_id, output_dir, raise_on_line_failure)
-            return batch_result
+                loop = asyncio.get_event_loop()
```

Comment:
this line will raise RuntimeError: There is no current event loop in thread 'MainThread'.

Reply:
Updated.